src/models/selection.ts

```diff
@@ -1,15 +1,23 @@
 import { RunGraphNode, Artifact } from '@/models'
 
 export type NodeSelection = {
-  id: string,
   kind: RunGraphNode['kind'],
+  id: string,
 }
 
 export type ArtifactSelection = {
   kind: 'artifact',
   id: string,
 }
 
-export type GraphItemSelection = NodeSelection | ArtifactSelection
+export type ArtifactClusterSelection = {
+  kind: 'artifactCluster',
+  ids: string[],
+}
+
+export type GraphItemSelection =
+  | NodeSelection
+  | ArtifactSelection
+  | ArtifactClusterSelection
 
-export type SelectableItem = RunGraphNode | Artifact
\ No newline at end of file
+export type SelectableItem = RunGraphNode | Artifact | string[]
```

Comment:
What is the `string[]`? Multiple ids of artifacts? That might get confusing once we have multiple events selected as well.